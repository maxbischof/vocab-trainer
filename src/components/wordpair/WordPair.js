import React from 'react'
import styled from 'styled-components'
import { ReactComponent as DeleteSVG } from '../../icons/delete.svg'

export default function WordPair({ word, deleteWord, index }) {
  return (
    <StyledWordpair key={word.foreign}>
      <Words>
        <b>{word.foreign}</b>
        <small>{word.native}</small>
      </Words>
      <DeleteIcon onClick={() => deleteWord(index)} />
    </StyledWordpair>
  )
}

const StyledWordpair = styled.li`
  padding: 10px;
  border-bottom: 1px solid var(--text);
  display: grid;
  grid-template-columns: 1fr auto;
  align-items: center;

  :last-child {
    border-bottom: none;
  }
`

const Words = styled.div`
  grid-column: 1 / 2;
  display: flex;
  flex-direction: column;
`

const DeleteIcon = styled(DeleteSVG)`
  grid-column: 2 / 3;
  grid-row: 1 / 3;
  display: grid;
  align-items: center;
  margin: 0 0 0 25px;
`
