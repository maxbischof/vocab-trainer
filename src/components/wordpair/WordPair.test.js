import React from 'react'
import WordPair from './WordPair'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const wordpair = { foreign: 'le ordinateur', native: 'der Computer' }
  const tree = renderer.create(<WordPair word={wordpair} />).toJSON()
  expect(tree).toMatchSnapshot()
})
