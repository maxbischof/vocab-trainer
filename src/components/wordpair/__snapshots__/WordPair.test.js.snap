// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li
  className="sc-bdfBwQ ftQkNA"
>
  <div
    className="sc-gsTCUz eugYRC"
  >
    <b>
      le ordinateur
    </b>
    <small>
      der Computer
    </small>
  </div>
  <svg
    className="sc-dlfnbm lnKTpL"
    onClick={[Function]}
  >
    delete.svg
  </svg>
</li>
`;
